### ISSUE TITLE
`btn-group` has inconsistent focus/active states when combined with hover states

### ISSUE THREAD

**Issue by umpox** [twbs/bootstrap#34106]
Code example taken [directly from the docs](https://getbootstrap.com/docs/5.0/components/button-group/#basic-example):

```
<div class="btn-group" role="group" aria-label="Basic example">
  <button type="button" class="btn btn-primary">Left</button>
  <button type="button" class="btn btn-primary">Middle</button>
  <button type="button" class="btn btn-primary">Right</button>
</div>
```

Left button active, middle button hovered:
![image](https://user-images.githubusercontent.com/9516420/119473453-15bb7f80-bd43-11eb-8224-908d63f28342.png)

Middle button active, left button hovered:
![image](https://user-images.githubusercontent.com/9516420/119473578-35eb3e80-bd43-11eb-8977-167d3572eed8.png)

Operating system: macOS Catalina 10.15.7
Browser: Chrome 90.0.4430.212

**Comment by alecpl** [comment]
I guess this could be fixed by using `z-index:2` on focused button, and `z-index:1` on hovered button.

**Comment by ffoodd** [comment]
Agreed with the diagnosis, we need to decouple both states' `z-index` here.

**Comment by TobyKurniawan** [comment]
hey, if anyone's not working on this I can give this a shot - would like to try to start contributing :)
